src/data/finset/basic.lean

```diff
@@ -402,6 +402,10 @@ lemma eq_empty_of_is_empty [is_empty α] (s : finset α) : s = ∅ :=
 finset.eq_empty_of_forall_not_mem is_empty_elim
 
 /-! ### singleton -/
+
+section singleton
+variables {s : finset α} {a : α}
```

Comment:
I think it would be better to inline these into your two new lemmas, so as to match the surrounding lemmas.

Personally, I prefer the data variables in the lemma anyway, as it's easier to check if the explicitness is correct.

Reply:
Files with implicit variables pulled out to the top of the section are much more readable in my opinion, and shorter as well because that usually lets you give one or two lines less **per statement**. I do not touch all lemmas in the section because I do not want to scope creep, but I at least try to not make the matter worse. And the next PR won't have to add the `variables`.

Reply:
I don't think saving lines is a particularly high priority goal, especially when those lines aren't the proofs themselves.

I'd prefer to keep this section consistent; meaning adding these variables to the lemmas directly for now, and making a new PR to pull everything to variables if you think it's worth the effort and want to paint some bikesheds.

Reply:
Meh